internal/auth/oidc/repository_auth_method_delete_test.go

```diff
@@ -0,0 +1,87 @@
+package oidc
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/hashicorp/boundary/internal/db"
+	"github.com/hashicorp/boundary/internal/errors"
+	"github.com/hashicorp/boundary/internal/iam"
+	"github.com/hashicorp/boundary/internal/kms"
+	"github.com/hashicorp/boundary/internal/oplog"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestRepository_DeleteAuthMethod(t *testing.T) {
+	conn, _ := db.TestSetup(t, "postgres")
+	rw := db.New(conn)
+	wrapper := db.TestWrapper(t)
+	kmsCache := kms.TestKms(t, conn, wrapper)
+	ctx := context.Background()
+
+	tests := []struct {
+		name            string
+		authMethod      *AuthMethod
+		wantRowsDeleted int
+		wantErrMatch    *errors.Template
+	}{
+		{
+			name: "valid",
+			authMethod: func() *AuthMethod {
+				org, _ := iam.TestScopes(t, iam.TestRepo(t, conn, wrapper))
+				databaseWrapper, err := kmsCache.GetWrapper(context.Background(), org.PublicId, kms.KeyPurposeDatabase)
+				require.NoError(t, err)
+				return TestAuthMethod(t, conn, databaseWrapper, org.PublicId, InactiveState, TestConvertToUrls(t, "https://alice.com")[0], "alice_rp", "alices-dogs-name")
+			}(),
+			wantRowsDeleted: 1,
+		},
+		{
+			name:         "no-public-id",
+			authMethod:   func() *AuthMethod { am := AllocAuthMethod(); return &am }(),
+			wantErrMatch: errors.T(errors.InvalidPublicId),
```

Comment:
What is the reason for not adding WantRowsDeleted: 0 here?

Reply:
I'm not always consistent, but I try not to add zero values explicitly for tests.  Maybe that's not the best choice?

Reply:
I am a fan of not explicitly adding zero values